contracts/minter-suite/Minters/MinterDAExpRefund/MinterDAExpRefundV0.sol

```diff
@@ -0,0 +1,965 @@
+    }
+
+    /**
+     * @notice Refunds the sender's payments above current amount due for
+     * projects in `_projectIds`. The amount due is the the price paid for
+     * the most recently purchased token on each project, or the base price if
+     * the artist has withdrawn revenues after the auction reached base price.
+     * This function is callable at any point, but is expected to typically be
+     * called after auctions have sold out above base price, or after auctions
+     * have been purchased at base price. This minimizes the amount of gas
+     * required to refund the sender.
+     * Entire transaction reverts if any refund fails.
+     * Sends total of all refunds to address `_to` in a single chunk.
+     * @param _to Address to send refund to.
+     * @param _projectIds Array of project IDs to refund payments on.
+     */
+    function claimRefundsTo(address payable _to, uint256[] memory _projectIds)
+        public
+        nonReentrant
+    {
+        // CHECKS
+        // input validation
+        require(_to != address(0), "No refund to the zero address");
+        // EFFECTS
+        // for each project, tally up the refund amount and update the receipt
+        // in storage
+        uint256 refund;
+        uint256 projectIdsLength = _projectIds.length;
+        for (uint256 i = 0; i < projectIdsLength; ) {
+            uint256 projectId = _projectIds[i];
+            ProjectConfig storage _projectConfig = projectConfig[projectId];
+            Receipt storage receipt = receipts[msg.sender][projectId];
+            // input validation
+            // require that a user has purchased at least one token on this project
+            require(
+                receipt.numPurchased > 0,
+                "No purchases made by this address"
+            );
+            // get the latestPurchasePrice, which returns the sellout price if the
+            // auction sold out before reaching base price, or returns the base
+            // price if auction has reached base price and artist has withdrawn
+            // revenues.
+            // @dev if user is elligible for a refund, they have purchased a token,
+            // therefore we are guaranteed to not have a populated
+            // latestPurchasePrice
+            uint256 netTokenPrice = _projectConfig.latestPurchasePrice;
+            // calculate the refund amount
+            // implicit overflow/underflow checks in solidity ^0.8
+            uint256 amountDue = receipt.numPurchased * netTokenPrice;
+            refund += receipt.netPaid - amountDue;
+            // reduce the netPaid (in storage) to value after refund deducted
+            receipt.netPaid = amountDue;
+            // emit event indicating new receipt state
+            emit ReceiptUpdated(
+                msg.sender,
+                projectId,
+                receipt.numPurchased,
+                receipt.netPaid
+            );
+            // gas efficiently increment i
```

Comment:
Nit: may want to explain that this is safe because we run out of gas way before we overflow

Reply:
good call, updated!